spec/screenPicker.spec.ts

```diff
@@ -0,0 +1,319 @@
+        });
+
+        it('should call `handleKeyUpPress` down arrow key correctly', () => {
+            const spy = jest.spyOn(ScreenPicker.prototype, 'setState');
+            const expectedValue = { selectedSource: { display_id: '0', id: '0', name: 'Application screen 0', thumbnail: undefined }};
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            wrapper.setState(stateMock);
+            events.keyup(keyCode.downArrow);
+            expect(spy).lastCalledWith(expectedValue);
+        });
+
+        it('should call `handleKeyUpPress` up arrow key correctly', () => {
+            const spy = jest.spyOn(ScreenPicker.prototype, 'setState');
+            const expectedValue = { selectedSource: { display_id: '2', id: '2', name: 'Application screen 2', thumbnail: undefined }};
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            wrapper.setState(stateMock);
+            events.keyup(keyCode.upArrow);
+            expect(spy).lastCalledWith(expectedValue);
+        });
+
+        it('should call `handleKeyUpPress` enter key correctly', () => {
+            const spy = jest.spyOn(ipcRenderer, sendEventLabel);
+            const expectedValue = { display_id: '1', id: '1', name: 'Application screen 1', thumbnail: undefined };
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            wrapper.setState(stateMock);
+            events.keyup(keyCode.enterKey);
+            expect(spy).lastCalledWith(screenSourceSelectedLabel, expectedValue);
+        });
+
+        it('should call `handleKeyUpPress` escape key correctly', () => {
+            const spy = jest.spyOn(ipcRenderer, sendEventLabel);
+            const expectedValue = { cmd: 'close-window', windowType: 'screen-picker' };
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            wrapper.setState(stateMock);
+            events.keyup(keyCode.escapeKey);
+            expect(spy).nthCalledWith(1, screenSourceSelectedLabel, null);
+            expect(spy).nthCalledWith(2, symphonyApiLabel, expectedValue);
+        });
+
+        it('should call `handleKeyUpPress` end key correctly', () => {
+            const spy = jest.spyOn(ScreenPicker.prototype, 'setState');
+            const expectedValue = { selectedSource: { display_id: '0', id: '0', name: 'Application screen 0', thumbnail: undefined }};
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            wrapper.setState(stateMock);
+            events.keyup(keyCode.endKey);
+            expect(spy).lastCalledWith(expectedValue);
+        });
+
+    });
+
+    describe('tab titles', () => {
+        it('should show `application-tab` when source name is not Entire screen', () => {
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            const applicationScreenStateMock = { sources: [{ display_id: '1', id: '1', name: 'Application screen', thumbnail: undefined }]};
```

Comment:
The use case where `Screen 2` is not at the beginning of the source name should not display the `screen tab`
```suggestion
            const applicationScreenStateMock = { sources: [
                   { display_id: '1', id: '1', name: 'Application screen', thumbnail: undefined }
                   { display_id: '2', id: '2', name: 'Application Screen 2', thumbnail: undefined }
            ]};
```

Reply:
:+1: